src/utils/elements/Ellipse.ts

```diff
@@ -0,0 +1,183 @@
+import { Size } from '../../types'
+import {
+  angle,
+  distance,
+  isEqual,
+  orientation,
+  orthogonal,
+  rotate,
+  times,
+  unitVector,
+  vector,
+} from '../maths'
+
+import GraphElement from './GraphElement'
+import Point, { PointXY } from './Point'
+import Segment, { SegmentAB } from './Segment'
+
+export default abstract class Ellipse extends GraphElement {
```

Comment:
We just need one implementation of the ellipse, based on the data model we are using. No need for an abstract class + many implementations